clash-lib/src/Clash/Backend/SystemVerilog.hs

```diff
@@ -358,6 +361,12 @@ tyDec ty@(Product _ _ tys) | typeSize ty > 0 = Just A.<$> prodDec
         Just y' -> Just A.<$> (pure y' <+> x <> semi)
     tName    = tyName ty
     selNames = map (\i -> tName <> "_sel" <> int i) [0..]
+tyDec ty@(MemBlob n m) | typeSize ty > 0 = Just A.<$>
+  "typedef" <+> "logic"
+            <+> brackets (int (m - 1) <> colon <> int 0)
+            <+> tyName ty
+            <+> brackets (int 0 <> colon <> int (n - 1))
+            <> semi
```

Comment:
I would remove this and instead add the following to `normalizeType`: https://github.com/clash-lang/clash-compiler/blob/d087788ded8721c60a912b05015bdb28e5d36aaa/clash-lib/src/Clash/Backend/SystemVerilog.hs#L292

```haskell
normaliseType (MemBlob n m) = Vector n (BitVector m)
```